File: code_learning/牛客/contest/2024-12-15/B.py
ss = input()
a = []
for i in ss:
    a.append(int(i))
b = a[0]
cnt, ans = 1, 0
for i in range(1, len(a)):
    if a[i] != b:
        cnt += 1
    elif cnt >= 2:
        ans += cnt * (cnt + 1) // 2 - cnt
        cnt = 1
    b = a[i]
if cnt >= 2:
    ans += cnt * (cnt + 1) // 2 - cnt
print(ans)

File: code_learning/牛客/contest/2024-12-15/B.cpp
#include <bits/stdc++.h>
using i64 = long long;

void solve() {
    std::string s; std::cin >> s;
    int n = s.size();
    std::vector<int> a(n);
    for (int i = 0; i < n; i++) {
        a[i] = s[i] - '0';
    }
    i64 b = a[0], cnt = 1, ans = 0;
    for (int i = 1; i < n; i++) {
        if (a[i] != b) cnt++;
        else if (cnt >= 2) {
            ans += (cnt * (cnt - 1) >> 1); cnt = 1;
        }
        b = a[i];
    }
    if (cnt >= 2) ans += (cnt * (cnt - 1) >> 1);
    std::cout << ans << '\n';
}

int main() {
    std::cin.tie(nullptr)->sync_with_stdio(false);

    solve();

    return 0;
}